rust/main/agents/relayer/src/relayer.rs

```diff
@@ -620,7 +621,277 @@ impl Relayer {
         }))
         .instrument(span)
     }
+
+    /// Helper function to build and return a hashmap of mailboxes.
+    /// Any chains that fail to build mailbox will not be included
+    /// in the hashmap. Errors will be logged and chain metrics
+    /// will be updated for chains that fail to build mailbox.
+    pub async fn build_mailboxes(
+        settings: &RelayerSettings,
+        core_metrics: &CoreMetrics,
+        chain_metrics: &ChainMetrics,
+    ) -> HashMap<HyperlaneDomain, Arc<dyn Mailbox>> {
+        settings
+            .build_mailboxes(settings.destination_chains.iter(), &core_metrics)
+            .await
+            .into_iter()
+            .filter_map(|(origin, mailbox_res)| match mailbox_res {
+                Ok(mailbox) => Some((origin, mailbox)),
+                Err(err) => {
+                    let message = err.to_string();
+                    error!(?err, origin=?origin, "{message}");
```

Comment:
Since we are going to have error serialised into a field in logs, would it make sense to give a more high level error message here, for example, an error message which will be independent from the internal error so that we can easily search by it in the logs?

Reply:
agreed, can you have the log message be a literal string like "Critical error when building mailbox" or something? here and in the validator announce one. Having `?err` is enough for us

Reply:
sounds good!